File: src/main/kotlin/com/example/whykotlin/java/extensionfunctions/ExtendSomething.java
package com.example.whykotlin.java.extensionfunctions;

import java.util.List;

import static java.util.Arrays.asList;

public class ExtendSomething {

    private void swap(List<Integer> list, Integer index1, Integer index2) {
        Integer tmp = list.get(index1);
        list.set(index1, list.get(index2));
        list.set(index2, tmp);
    }

    void swapSomething() {
        List<Integer> list = asList(1, 2, 3);
        swap(list, 0, 2);
    }
}

File: src/main/kotlin/com/example/whykotlin/kotlin/extensionfunctions/ExtendSomething.kt
package com.example.whykotlin.kotlin.extensionfunctions

class ExtendSomething {

    private fun MutableList<Int>.swap(index1: Int, index2: Int) {
        val tmp = this[index1]
        this[index1] = this[index2]
        this[index2] = tmp
    }

    fun swapSomething() {
        val list = mutableListOf(1, 2, 3)
        list.swap(0, 2)
    }
}